src/readcalls.c

```diff
@@ -197,206 +195,30 @@ int readcalls(const char *logfile) {
 
-	    worked[station].band |= inxes[bandindex];	/* mark band as worked */
-
-	    qsos_per_band[bandindex]++;
-
-	    if (CONTEST_IS(CQWW) || itumult || wazmult)
-		zones[z] |= inxes[bandindex];
-
-	    if (pfxnumcntidx < 0) {
-		if (!veto) {
-		    countries[countrynr] |= inxes[bandindex];
-		}
-	    } else {
-		pfxnummulti[pfxnumcntidx].qsos[pxnr] |= inxes[bandindex];
-	    }
-	}
 	free_qso(qso);
 	qso = NULL;
     }
 
     fclose(fp);
 
-    /* all lines red, now build other statistics */
-    if (CONTEST_IS(WPX) || pfxmult) {
-	char checkcall[20];
-
-	/* build prefixes_worked array from list of worked stations */
-	InitPfx();
-
-	for (int n = 0; n < nr_worked; n++) {
-	    strcpy(checkcall, worked[n].call);
-	    getpx(checkcall);
-	    /* FIXME: wpx is counting pfx only once so bandindex is not
-	     * really needed here. If you have wpx and pfxmultab set the
-	     * count for the last read band wil be wrong as all pfx will be
-	     * counted for that band.
-	     * Maybe better use BANDINDEX_OOB here:
-	     * - Will count pfx for wpx correctly
-	     * - but will not change counts for pfxmultab on contest bands */
-	    add_pfx(wpx_prefix, BANDINDEX_OOB);
-	}
-    }
-
-    if (CONTEST_IS(CQWW) || itumult || wazmult) {
-	for (int n = 1; n < MAX_ZONES; n++) {
-	    count_contest_bands(zones[n], zonescore);
-	}
-    }
-
-    if (CONTEST_IS(CQWW)) {
-	for (int n = 1; n <= MAX_DATALINES - 1; n++) {
-	    count_contest_bands(countries[n], countryscore);
-	}
-    }
-
-    if (dx_arrlsections) {
-	for (int cntr = 1; cntr < MAX_DATALINES; cntr++) {
-	    if (cntr != w_cty && cntr != ve_cty) {	// W and VE don't count here...
-		count_contest_bands(countries[cntr], countryscore);
-	    }
-	}
-    }
-
-    if (CONTEST_IS(ARRLDX_USA)) {
-	for (int cntr = 1; cntr < MAX_DATALINES; cntr++) {
-	    if (cntr != w_cty && cntr != ve_cty) {	// W and VE don't count here...
-		count_contest_bands(countries[cntr], countryscore);
-	    }
-	}
-    }
-
-    if (CONTEST_IS(PACC_PA)) {
-	for (int n = 1; n < MAX_DATALINES; n++) {
-	    count_contest_bands(countries[n], countryscore);
-	}
-    }
-
-    if (country_mult || pfxnummultinr > 0) {
-
-	for (int n = 1; n <= MAX_DATALINES - 1; n++) {
-
-	    pfxnumcntidx = -1;
-	    if (pfxnummultinr > 0) {
-		pfxnumcntidx = lookup_country_in_pfxnummult_array(n);
-	    }
-
-	    if (pfxnumcntidx >= 0) { /* found in array */
-		/* count all possible pfx numbers
-		 * eg: K0, K1, K2, ..., K9 */
-		for (int pfxnr = 0; pfxnr < 10; pfxnr++) {
-		    count_contest_bands(pfxnummulti[pfxnumcntidx].qsos[pfxnr],
-					countryscore);
-		}
-	    } else {
-		// simple 'country_mult', but works together with pfxnummulti
-		count_contest_bands(countries[n], countryscore);
-	    }
-	}
-    }
-
     return linenr;			// nr of lines in log
 }
 
 int log_read_n_score() {
-    int nr_qsolines;
+    int nr_qsolines = readcalls(logfile);
 
-    total = 0;
-    nr_qsolines = readcalls(logfile);
     if (qtcdirection > 0) {
 	readqtccalls();
```

Comment:
On startup `readqtccalls()` is called before `readcalls()`. Hopefully the order is not relevant.

Reply:
Not quite sure. That needs to be checked.

Reply:
See issue #310